src/jobs/blast/components/BlastForm.tsx

```diff
@@ -149,6 +157,49 @@ const BlastForm = ({ initialFormValues }: Props) => {
     formValues[BlastFields.database].selected
   );
 
+  useEffect(() => {
+    const fetchData = async () => {
+      if (!excludeTaxonField && formValues[BlastFields.taxons].selected) {
+        const baseUrl = joinUrl(apiPrefix, 'uniprotkb', 'search');
+        let query = `(${(formValues[BlastFields.taxons].selected as SelectedTaxon[]).map((taxon) => `organism_id:${taxon.id}`).join(' OR ')})`;
+        switch (formValues[BlastFields.database].selected) {
+          case 'uniprotkb_refprotswissprot':
+            query += ' AND (keyword:KW-1185 AND reviewed:true)';
+            break;
+          case 'uniprotkb_reference_proteomes':
+            query += ' AND (keyword:KW-1185)';
+            break;
+          case 'uniprotkb_swissprot':
+            query += ' AND (reviewed:true)';
+            break;
+          case 'uniprotkb_pdb':
+            query += ' AND (structure_3d:true)';
+            break;
+          default:
+            break;
+        }
+        const url = stringifyUrl(baseUrl, {
+          format: fileFormatToUrlParameter[FileFormat.fasta],
+          includeIsoform: true,
+          query,
+        });
+        const headResponse = await fetch(url, {
```

Comment:
We could use `fetchData` and have a clean up for the `useEffect` which aborts eg:

```
    return () => {
      abortController.abort();
    };
```

`fetchGoCamModels` has an example of this.